### PR TITLE
Move JS from HAML file in views to JS file in assets/javascripts.

### PR DESCRIPTION
**Pull request by ashercodeorg**
In doing so, cleans up indentation and adds comments.

### CODE REVIEW

File: dashboard/app/assets/javascripts/levelCompletions.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
+// Shows the samplingMessage and completionTable DOM elements depending on the
+// parameters.
+//
+// isSampled: A boolean indicating whether Google Analytics sampled.
+// startDatePresent: A boolean indicating whether a start_date parameter was
+//     given.
```
Comment (Hamms) [inline dashboard/app/assets/javascripts/levelCompletions.js]: Might as well make this (and other comments) actual JSDoc comments:

```
/**
 * Shows the samplingMessage and completionTable DOM elements depending on the
 * parameters.
 *
 * @param {boolean} isSampled - whether Google Analytics sampled
 * @param {boolean} startDatePresent - whether a start_date parameter was given
 */
function hideAndShowDomElements(isSampled, startDatePresent) {
```

Comment (ashercodeorg) [inline dashboard/app/assets/javascripts/levelCompletions.js]: Done.

**Comment by Hamms** [comment]
I like having javascript in a JS file, but I don't know how I feel about including admin-specific code in ALL dashboard views, particularly since that includes all of code studio.

I don't know a ton about the rails asset build pipeline, but presumably we should be able to put this somewhere like `dashboard/app/assets/javascripts/admin/levelCompletions.js` and then include it in the haml with 

``` haml
- content_for(:head) do
    = javascript_include_tag 'admin/levelCompletions'
```

 such that it will be available at `/admin/levels` but not elsewhere?

**Comment by ashercodeorg** [comment]
Done. Keep the good suggestions coming.

File: dashboard/app/views/admin_reports/level_completions.html.haml [hunk 2]
```diff
@@ -28,131 +30,7 @@
...
-  $(function() {
-    if(#{ @is_sampled }) {
-      $('#samplingMessage').show();
-    }
-    if(#{ params[:start_date].present? }) {
-      $('#completionTable').show();
-    }
-  });
+  var headers = eval(#{ headers.try(:to_json)});
   var data = eval(#{ data.try(:to_json)});
```
Comment (Hamms) [inline dashboard/app/views/admin_reports/level_completions.html.haml]: is `eval` necessary for either of these?

File: dashboard/app/assets/javascripts/admin/levelCompletions.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
+/**
+ * Shows the samplingMessage and completionTable DOM elements depending on the
+ * parameters.
+ *
+ * @param {boolean}: isSampled - whether Google Analytics sampled.
+ * @param {boolean} startDatePresent - whether a start_date parameter was given.
+ */
+//
```
Comment (Hamms) [inline dashboard/app/assets/javascripts/admin/levelCompletions.js:8]: superminor nit: stray empty comment.

Also, `@param {boolean}: isSampled` above shouldn't have the `:`

**Comment by Hamms** [comment]
Awesome! I'm glad that actually worked, although slightly paranoid that the production build pipeline might be different enough that it works in development but not prod.

I'm not paranoid enough that I think it's worth the time to manually test it a production-like environment, but I am paranoid enough that we should definitely manually verify it works as expected when it hits prod.

Aside from a couple of minor nits that you're free to ignore, this LGTM!

File: dashboard/app/views/admin_reports/level_completions.html.haml (hunk 2 above)
Comment (ashercodeorg) [inline dashboard/app/views/admin_reports/level_completions.html.haml]: Seemingly not. Removed.

File: dashboard/app/assets/javascripts/admin/levelCompletions.js (hunk 3 above)
Comment (ashercodeorg) [inline dashboard/app/assets/javascripts/admin/levelCompletions.js:8]: Done.

**Pull request merged**